DOC: Clarify where to the additional arguments for some win_types

linnndachen opened the pull request:
For example, std needs to specify when win_types is gaussian. However, std should be specified in the operation argument, not as one of the rolling arguments. This change is to clarify this point. 

- [ ] closes #34593
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

linnndachen committed:
DOC: Clarify some syntax when using win_types

DOC: Clarify where to the additional arguments for some win_types
For example, std needs to specify when win_types is gaussian. However, std should be specified in the operation argument, not as one of the rolling arguments. This change is to clarify this point. 

Closes: #34593

dsaxton commented on pandas/core/window/rolling.py:
```diff
@@ -922,11 +922,11 @@ class Window(_Window):
+    * ``gaussian`` (needs std in operation argument)
+    * ``general_gaussian`` (needs power, width in operation argument)
+    * ``slepian`` (needs width in operation argument)
+    * ``exponential`` (needs tau in operation argument), center is set to None.
```
I'm wondering if it may be helpful to also add some additional explanation below (something like: "Certain window types require that a scale parameter be passed...") since this is a fairly confusing situation, and then call attention to the Gaussian case in the examples section. Thoughts @wchen928 @MarcoGorelli?

wchen928 replied:
How about we change to the following and add one more sentence before we list all the examples. 

```suggestion
    * ``exponential`` (need parameter: tau), center is set to None.
```
Additional sentence: Certain window types require a scale parameter to be passed. Please see the third example below to see how to add the additional parameters.

MarcoGorelli replied:
> Please see the third example below on how to add the additional parameters.

Sure, sounds good to me

wchen928 replied:
Great. I will wait for @dsaxton 's opinion and before editing it.

dsaxton replied:
Works for me too 👍

linnndachen committed:
DOC: Clarify where to the additional arguments for some win_types

Edit: 711add5 First Commit
Original issue: #34615

pep8speaks commented:
Hello @wchen928! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/window/rolling.py`](https://github.com/pandas-dev/pandas/blob/a3e38ac24cd7b8ba89d231541818e1aa6c77fc68/pandas/core/window/rolling.py):

> [Line 934:1](https://github.com/pandas-dev/pandas/blob/a3e38ac24cd7b8ba89d231541818e1aa6c77fc68/pandas/core/window/rolling.py#L934): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 935:89](https://github.com/pandas-dev/pandas/blob/a3e38ac24cd7b8ba89d231541818e1aa6c77fc68/pandas/core/window/rolling.py#L935): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 88 characters)
> [Line 935:100](https://github.com/pandas-dev/pandas/blob/a3e38ac24cd7b8ba89d231541818e1aa6c77fc68/pandas/core/window/rolling.py#L935): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

linnndachen committed:
DOC: Clarify where to the additional arguments for some win_types

Edit: a3e38ac Second Commit 

What's new: I shortened line 935 so that each line is less than 88 characters. 

Original Issue: #34615

linnndachen committed:
Revert "DOC: Clarify where to the additional arguments for some win_types"

This reverts commit 45be5386b656650443404fae1842f169a65a8919.

linnndachen committed:
Revert "Revert "DOC: Clarify where to the additional arguments for some win_types""

This reverts commit 05ed0eb01dcc0a4ec4743f30ba411c9fa6ee402f.

linnndachen committed:
DOC: Remove whitespace in docstring

linnndachen committed:
DOC: Remove indent of line934

WillAyd commented on pandas/core/window/rolling.py:
```diff
@@ -922,16 +922,19 @@ class Window(_Window):
     different window types see `scipy.signal window functions
     <https://docs.scipy.org/doc/scipy/reference/signal.html#window-functions>`__.
 
+    Certain window types require a scale parameter to be passed. Please see
```
Is this supposed to read `scalar parameter` or is there an actual `scale` argument you are saying needs to be passed?

Minor but once clarified this is good to go

MarcoGorelli replied:
Yes, as far as I understand, `power` isn't a scale parameter for the general gaussian distribution - `scalar parameter` or even just `parameter` might be better

wchen928 replied:
```suggestion
    Certain window types require additional parameters to be passed. Please see
```

linnndachen committed:
Update pandas/core/window/rolling.py

TomAugspurger committed:
Merge remote-tracking branch 'upstream/master' into patch-2

TomAugspurger commented:
Merged master to fix the CI failure.

mroeschke commented:
Thanks @wchen928
